src/safeposix/syscalls/sys_calls.rs

```diff
@@ -222,8 +238,9 @@ impl Cage {
             // newsigset.insert(0, mainsigset);
         }
 
+        // Update status
         let newcage = Cage {
```

Comment:
we can talk a bit more about this on Monday but I believe that we don't need to create a new cage struct here, the old one should be sufficient